services/task-svc/internal/task/task.go

```diff
@@ -0,0 +1,240 @@
+package task
+
+import (
+	"context"
+	"github.com/google/uuid"
+	zlog "github.com/rs/zerolog/log"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"hwgorm"
+	"task-svc/api"
+	intPatient "task-svc/internal/patient"
+)
+
+type Base struct {
+	Name        string `gorm:"column:name"`
+	Description string `gorm:"column:description"`
+	Status      uint32 `gorm:"column:status"`
+}
+
+type Task struct {
+	Base
+	Id             uuid.UUID     `gorm:"column:id"`
+	AssignedUserId uuid.NullUUID `gorm:"column:assigned_user_id;default:NULL"`
+	PatientId      uuid.UUID     `gorm:"column:patient_id"`
+}
+
+type ServiceServer struct {
+	api.UnimplementedTaskServiceServer
+}
+
+func NewServiceServer() *ServiceServer {
+	return &ServiceServer{}
+}
+
+func (s ServiceServer) CreateTask(ctx context.Context, req *api.CreateTaskRequest) (*api.CreateTaskResponse, error) {
+	log := zlog.Ctx(ctx)
+	db := hwgorm.GetDB(ctx)
+
+	// TODO: Auth
+
+	patientId, err := uuid.Parse(req.PatientId)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	// Check if patient exists
+	if err := db.First(intPatient.Patient{Id: patientId}).Error; err != nil {
+		return nil, status.Error(codes.NotFound, err.Error())
+	}
+
+	task := Task{
+		Base: Base{
+			Name:        req.Name,
+			Description: req.Description,
+		},
+		PatientId: patientId,
+	}
+
+	result := db.Create(&task)
+	if err := result.Error; err != nil {
+		log.Warn().Err(err).Msg("database error")
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	log.Info().
+		Str("taskId", task.Id.String()).
+		Str("patientId", patientId.String()).
+		Msg("task created for patient")
+
+	return &api.CreateTaskResponse{
+		Id: task.Id.String(),
+	}, nil
+}
+
+func (ServiceServer) GetTask(ctx context.Context, req *api.GetTaskRequest) (*api.GetTaskResponse, error) {
+	log := zlog.Ctx(ctx)
+	db := hwgorm.GetDB(ctx)
+
+	// TODO: Auth
+
+	id, err := uuid.Parse(req.Id)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	task := Task{Id: id}
+	result := db.First(&task)
+	if err := result.Error; err != nil {
+		log.Warn().Err(err).Msg("database error")
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	assignedUserId := task.AssignedUserId.UUID.String()
+	if !task.AssignedUserId.Valid {
```

Comment:
is this needed? wont it be a zero-type else and NULLed by gorm

Reply:
It's the `uuid.NullUUID` handling. How should we handle this?

Reply:
Oh I though this would prepare a db query 😅

But now I think there is a different issue here:
If it can be null we should mark it as `optional` in the .proto, I think the go generator will then let us work with pointers, and we should use nil in this case

Reply:
In postgres a null uuid gets represented by `00000000-0000-0000-0000-000000000000`. The stringification of `uuid.NullUUID` should handle the translation of `00000000-0000-0000-0000-000000000000` to `""` but this seems not to work. I've added a code comment.